documentation/src/main/asciidoc/reference/search-dsl-predicate.asciidoc

```diff
@@ -630,6 +630,71 @@ but can be <<search-dsl-predicate-common-boost,boosted>>,
 either on a per-field basis with a call to `.boost(...)` just after `.field(...)`/`.fields(...)`
 or for the whole predicate with a call to `.boost(...)` after `.matching(...)`.
 
+[[search-dsl-predicate-terms]]
+== `terms`: match a set of terms
+
+The `terms` predicate matches documents for which a given field contains some terms, any or all of them.
+
+With `matchingAny` we require that at least one of the provided terms matches.
+Functionally, this is somewhat similar to a <<search-dsl-predicate-boolean-or,boolean `OR`>> with one <<search-dsl-predicate-match,`match`>> predicate per term, but the syntax for a single `terms` predicate is more concise.
+
+NOTE: `matchingAny` expects to be passed *terms*, not just any string. The given terms will not be analyzed. See <<search-dsl-predicate-terms-analysis>>.
+
+.Matching any of the provided terms
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=terms-any]
+----
+====
+
+With `matchingAll` we require that all of the provided terms match.
+Functionally, this is somewhat similar to a <<search-dsl-predicate-boolean-and,boolean `AND`>> with one <<search-dsl-predicate-match,`match`>> predicate per term, but the syntax for a single `terms` predicate is more concise.
+
+NOTE: `matchingAll` expects to be passed *terms*, not just any string. The given terms will not be analyzed. See <<search-dsl-predicate-terms-analysis>>.
+
+[WARNING]
+====
+With `matchingAll` the given terms must not be more than 1024.
+====
+
+.Matching all of the provided terms
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=terms-any]
+----
+====
+
```

Comment:
Since you added support for argument conversion, could you please add a section named "Expected type of arguments"? You can pretty much copy/paste the one from the `match` predicate (with a few changes). See [here](https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-dsl-predicate-match-argument-type).